src/modules/keyserver/key-server.usecase.ts

```diff
@@ -0,0 +1,35 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { UserAttributes } from '../user/user.domain';
+import { Keys } from './key-server.domain';
+import { KeyServerRepository } from './key-server.repository';
+
+@Injectable()
+export class KeyServerUseCases {
+  constructor(
+    @Inject('KEY_SERVER_REPOSITORY')
+    private readonly repository: KeyServerRepository,
+  ) {}
+
+  async addKeysToUser(userId: UserAttributes['id'], keys: Keys): Promise<Keys> {
+    if (!keys.publicKey) {
```

Comment:
These checks should be on the controller, not in the use case, as Keys are not an optional on the interface too, therefore the checks are not correct according to the types, which indicates that those params are not null or undefined

Reply:
Removed